types/recurly__recurly-js/lib/google-pay/native.d.ts

```diff
@@ -0,0 +1,336 @@
+   */
+  currencyCode?: string;
+
+  /**
+   * The status of the total price used:
+   */
+  totalPriceStatus?:
+    /**
+     * Total price might adjust based on the details of the response, such as sales tax collected that's based on a billing address.
+     */
+    | 'ESTIMATED'
+
+    /**
+     * Total price doesn't change from the amount presented to the shopper.
+     */
+    | 'FINAL';
+
+  /**
+   * Total cost to display in the Google Pay payment sheet.
+   */
+  totalPrice?: string;
+
+  /**
+   *  All of the available charges for the current payment request.
+   *  Required and only populated in the payment sheet if you use Authorize Payments or Dynamic Price Updates.
+   */
+  displayItems?: GooglePayDisplayItem[];
+
+  /**
+   * Label for the total price within the display items.
+   */
+  totalPriceLabel?: string;
+
+  /**
+   * Affects the submit button text displayed in the Google Pay payment sheet.
+   */
+  checkoutOption?:
+    /**
+     * Standard text applies for the given totalPriceStatus (default).
+     */
+    | 'DEFAULT'
+    /**
+     * The selected payment method is charged immediately after the payer confirms their selections.
+     * This option is only available when totalPriceStatus is set to FINAL.
+     */
+    | 'COMPLETE_IMMEDIATE_PURCHASE';
+};
+
+/**
+ * Provides information about any Offers currently applied to a transaction.
+ */
+export type GooglePayOfferDetail = {
+  /**
+   * The promotional code associated with the Offer. Should be the same as the code entered into the payment sheet.
+   */
+  redemptionCode: string;
+
+  /**
+   * A description of the promotion applied by the Offer code.
+   */
+  description: string;
+};
+
+/**
+ * Provides information about any Offers currently applied to a transaction.
+ */
+export type GooglePayOfferInfo = {
+  offers: GooglePayOfferDetail[];
+};
+
+/**
+ * Set shipping restrictions.
+ */
+export type GooglePayShippingAddressParameters = {
+   /**
+    * ISO 3166-1 alpha-2 country code values of the countries where shipping is allowed.
+    * If this object isn't specified, all shipping address countries are allowed.
+    */
+  allowedCountryCodes?: string[];
+
+  /**
+   * Set to true if a phone number is required for the provided shipping address.
+   */
+  phoneNumberRequired?: boolean;
+};
+
+/**
+ * Shipping option selection
+ */
+export type GooglePaySelectionOption = {
+  /**
+   * The developer can put any value that needs to be returned in PaymentData.
+   */
+  id: string;
+
+  /**
+   * The label to be displayed as the option.
+   */
+  label: string;
+
+  /**
+   * A descriptive text that is displayed below the option label.
+   */
+  description?: string;
+};
+
+/**
+ * Set shipping options.
+ */
+export type GooglePayShippingOptionParameters = {
+  /**
+   * All of the shipping options available for the current request.
+   */
+  shippingOptions: GooglePaySelectionOption[];
+
+  /**
+   * 	An identifier to the default selected shipping option.
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/recurly__recurly-js/lib/google-pay/native.d.ts;
  Line: 180;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.